javawrapper/device/iothub-client-javawrapper/DeviceClient.java

```diff
@@ -0,0 +1,130 @@
+import javaWrapper.Iothub_client_wrapperLibrary.HTTP_Protocol;
+import javaWrapper.Iothub_client_wrapperLibrary.IOTHUB_CLIENT_HANDLE;
+import javaWrapper.Iothub_client_wrapperLibrary.IOTHUB_CLIENT_RESULT;
+import javaWrapper.Iothub_client_wrapperLibrary.IOTHUB_CLIENT_TRANSPORT_PROVIDER;
+import javaWrapper.Iothub_client_wrapperLibrary.IotHubEventCallback;
+import javaWrapper.Iothub_client_wrapperLibrary.IotHubMessageCallback;
+import javaWrapper.Iothub_client_wrapperLibrary.MQTT_Protocol;
+
+
+public final class DeviceClient 
+{
+	private static IOTHUB_CLIENT_HANDLE handle = new IOTHUB_CLIENT_HANDLE(null);
+	private IOTHUB_CLIENT_TRANSPORT_PROVIDER transport;
+	
+	protected DeviceClient()
+	{
+	
+	}
+	
+	public enum IotHubClientProtocol 
+	{ 
+		HTTPS, AMQPS, MQTT 
+	} 
+
+	public DeviceClient(String connString, IotHubClientProtocol protocol)
+	{
+		switch (protocol) 
+		{ 
+			case HTTPS: 
+				this.transport = new HTTP_Protocol(); 
+		        break; 
+	        case AMQPS: 
+	        	this.transport = new AMQP_Protocol();  
+		        break; 
+		    case MQTT: 
+		    	this.transport = new MQTT_Protocol();
+		        break; 
+		    default: 
+		    // should never happen. 
+		        throw new IllegalStateException( 
+		       		"Invalid client protocol specified."); 
+		} 
+
+		handle = Iothub_client_wrapperLibrary.INSTANCE.IoTHubClient_CreateFromConnectionString(connString, transport);
+	}
+	
+	public void open()
+	{
+		if (handle == null)
+		{
+			throw new IllegalStateException(
+                    "Cannot open ");
+		}
+	}
+	
+	public void close()
+	{
+		if (handle == null)
+		{
+			throw new IllegalStateException(
+                    "Cannot close ");
+		}
+	}
+	
+	public void sendEventAsync(Message message,
+	            IotHubEventCallback callback,
+	            Pointer lockobj)
+	{
+		int status = IOTHUB_CLIENT_RESULT.IOTHUB_CLIENT_ERROR;
+		
+		if (handle == null)
+		{
+			throw new IllegalStateException(
+                    "Cannot send event from "
+                            + "an IoT Hub client that is closed.");
+		}
+			
+		status = Iothub_client_wrapperLibrary.INSTANCE.IoTHubClient_SendEventAsync(handle.getPointer(), message.getMessageHandle(), callback, lockobj);
+		
+		if (status != IOTHUB_CLIENT_RESULT.IOTHUB_CLIENT_OK)
+		{
+			throw new IllegalArgumentException("SendEventAsync error");
+		}
+	}
+	
+	
+	
+	public void setMessageCallback(IotHubMessageCallback messageCallback, Pointer userContextCallback)
+    {
+		Iothub_client_wrapperLibrary.INSTANCE.IoTHubClient_SetMessageCallback(handle.getPointer(), messageCallback, userContextCallback);
+    }
+	
+	public void setOption(String optionName, Object value)
+    {
+		int status = IOTHUB_CLIENT_RESULT.IOTHUB_CLIENT_ERROR;
+		
+		if(optionName == null)
+        {
+            throw new IllegalArgumentException("optionName is null");
+        }
+        else
+        {
+        	if (this.transport.getClass() == HTTP_Protocol.class)
```

Comment:
c will propagate this automatically to the proper protocol. No need to do this here.